python/setup.py

```diff
@@ -421,7 +421,8 @@ def _bundle_cython_cpp(self, name, lib_path):
     def _failure_permitted(self, name):
         if name == '_parquet' and not self.with_parquet:
             return True
-        if name == '_parquet_encryption' and not self.with_parquet_encryption:
+        if name == '_parquet_encryption' and \
+                not (self.with_parquet and self.with_parquet_encryption):
```

Comment:
From the name, looks to me `with_parquet_encryption == True` implies `with_parquet == True`, no?

Reply:
We could go that way, yes, and have one implicitly enable the other. We could also just fix the CI script not to have conflicting defaults: https://github.com/apache/arrow/blob/a32f454d86560ab357b419273bb08678e1632d30/ci/scripts/python_build.sh#L60-L61

Reply:
I fixed the script as well but IMO setup.py still either needs to 1) implicitly turn on Parquet if encryption is requested or 2) implicitly turn off encryption if Parquet is not enabled